sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -54,6 +58,8 @@ public RetryPolicy(int maxRetries, Duration delayDuration) {
         return next.clone().process()
             .flatMap(httpResponse -> {
                 if (shouldRetry(httpResponse, tryCount)) {
+                    logger.info("[Retrying] Try count: {}, HTTP response: {}, Original HTTP Request: {}",
```

Comment:
Let's move this into `determineDelayDuration` so we can also get the delay that will be used. Another option is setting a local variable to the delay duration and log it here.

Reply:
In addition, I don't think it is necessary to log the entire HTTP response and HTTP request. There is an HTTP pipeline policy that does this already, the HttpDetailLogLevel policy.

Outputting the retry attempt and the delay should be sufficient.